urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/othello/UIJuego.java

```diff
@@ -0,0 +1,427 @@
+			tableroUI.declararGanador(ganador);
+			tableroUI.notificarVictoria(controlador.obtenerGanador());
+		}
+	}
+	
+	public TipoTablero obtenerTipoTableroElegido(){
+		if (tableroUI.obtenerTipoTableroButton()[0].isSelected()) {
+			return TipoTablero.CLASICO;
+		} else if (tableroUI.obtenerTipoTableroButton()[1].isSelected()) {
+			return TipoTablero.OCTOGONAL;
+		} else {
+			return TipoTablero.PERSONALIZADO;
+		}
+	}
+	
+	public int obtenerTamTableroElegido(){
+		if (tableroUI.obtenerTamTableroButton()[0].isSelected()) {
+			return 10;
+		} else if (tableroUI.obtenerTamTableroButton()[1].isSelected()) {
+			return 12;
+		} else if (tableroUI.obtenerTamTableroButton()[2].isSelected()) {
+			return 14;
+		}else {
+			return 16;
+		}
+	}
+	
+	public void definirTipoTablero(int tamanio, TipoTablero tipoTablero){
+		tableroUI.dispose();
+		
+		if ( tipoTablero == TipoTablero.OCTOGONAL ) {
+			JFileChooser fileChooser = new JFileChooser();
+			String nombreFichero = "Octogonal10x10.oth";
+			if (tamanio == 12) {
+				nombreFichero = "Octogonal12x12.oth";
+			} else if (tamanio == 14) {
+				nombreFichero = "Octogonal14x14.oth";
+			} else if (tamanio == 16) {
+				nombreFichero = "Octogonal16x16.oth";
+			}
+			File directorio = new File("setup/boards");
+			File fichero = new File(directorio.getPath(),nombreFichero);
+			fileChooser.setCurrentDirectory(fichero);		
+			fileChooser.setFileFilter(miFiltro);
+			try {
+				leerArchivo(fichero, tipoTablero);
+				tableroUI = new TableroUI(Tablero.TABLERO_LARGO, Tablero.TABLERO_ANCHO, controlador.obtenerTableroLogica(), tipoTablero);
+				iniciarTableroUI();
+			} catch (FileNotFoundException e1) {
+				e1.printStackTrace();
+			}
+		}
+		
+		if ( tipoTablero == TipoTablero.CLASICO || tipoTablero == null) {
+			controlador.inicializar(tamanio, tamanio, tipoTablero);
+			tableroUI = new TableroUI(Tablero.TABLERO_LARGO, Tablero.TABLERO_ANCHO, controlador.obtenerTableroLogica(), tipoTablero);
+			iniciarTableroUI();
+		}
+		
+		if ( tipoTablero == TipoTablero.PERSONALIZADO ){
+			controlador.inicializar(tamanio, tamanio, tipoTablero);
+			tableroUI = new TableroUI(Tablero.TABLERO_LARGO, Tablero.TABLERO_ANCHO, controlador.obtenerTableroLogica(), tipoTablero);
+			iniciarTableroUI();
+			estaModificandoTablero = true;
+			tableroUI.mostrarBarraBoton();	
+		}
+		
+		run();
+	}
+	
+	private void escribirArchivo(File fichero) {
+		PrintWriter writer;
+		try {
+			int casillasJugablesIniciales=0;
+			writer = new PrintWriter(fichero);
+			Map<Point, EstadoCasilla> tableroActual = controlador.obtenerTableroLogica().obtenerTablero();
+			writer.println(TableroUI.largoTablero + "," + TableroUI.anchoTablero + "," + controlador.jugadorActual().color()
+					+ "," + obtenerTipoTableroElegido());
+			for(Point point : tableroActual.keySet()) {
+				writer.println(point.x + "," + point.y + "," + tableroActual.get(point).toString());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.